sale_commission/models/account_invoice.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2011 Pexego Sistemas Informáticos (<http://www.pexego.es>).
+#    Copyright (C) 2015 Avanzosc (<http://www.avanzosc.es>)
+#    Copyright (C) 2015 Pedro M. Baeza (<http://www.serviciosbaeza.com>)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields, api
+
+
+class AccountInvoice(models.Model):
+    """Invoice inherit to add salesman"""
+    _inherit = "account.invoice"
+
+    @api.one
+    @api.depends('invoice_line.agents.amount')
+    def _get_commission_total(self):
+        self.commission_total = 0.0
+        for line in self.invoice_line:
+            self.commission_total += sum(x.amount for x in line.agents)
+
+    commission_total = fields.Float(
+        string="Commissions", compute="_get_commission_total",
+        store=True)
+
+    @api.multi
+    def _refund_cleanup_lines(self, lines):
+        """ugly function to map all fields of account.invoice.line
+        when creates refund invoice"""
+        res = super(AccountInvoice, self)._refund_cleanup_lines(lines)
+        for line in res:
+            if 'commission_ids' in line[2]:
+                commission_ids = [(6, 0, line[2].get('commission_ids', []))]
```

Comment:
nitpicking: you don't need get here, you ensured the key was there in the previous line.